### DISCUSSION TITLE
How to reuse the common arguments between multi commands?

### DISCUSSION THREAD

**Discussion by dengyakui** [Tyrrrz/CliWrap#165] [Q&A] [answered]
### This tool is Awesome for invoke command like git. Respect to your works!
And I had a little problem in common use:

When I invoke multi git commands  with cli wrap, there are some  common args  to setup before each command run, 
like `WithWorkingDirectory`, `With Credentials`, `With Evnironment`...,  I wonder  is there some settings to share these args between commands , so we do not need to provide these args every time.

Or,  I was wonder should I to generate a temp script file first, then use CliWrap to inovke the script, is there some samples to execute multiline script file?
![image](https://user-images.githubusercontent.com/3945919/185527268-84efd2eb-eda9-46ce-9cf0-a7309e7983ed.png)

**Comment by Tyrrrz** [comment, marked as answer]
Extension methods. Either for `Command` or for `ArgumentsBuilder`/`EnvironmentVariablesBuilder`.

**Reply by dengyakui** [reply]
Thanks . I think I had resolved  by setting up a basic command obj. then and reuse it .
